$NetBSD: patch-configure,v 1.4 2011/12/29 16:19:25 rumko Exp $

Fix unportable test(1) construct.
Accepted by upstream, will be in next release.

Check for libm so it is included if need be.

--- configure.orig	2011-12-29 13:09:11.633992000 +0000
+++ configure
@@ -14689,6 +14689,51 @@ fi
 
 done
 
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for sqrt in -lm" >&5
+$as_echo_n "checking for sqrt in -lm... " >&6; }
+if ${ac_cv_lib_m_sqrt+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-lm  $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char sqrt ();
+int
+main ()
+{
+return sqrt ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_lib_m_sqrt=yes
+else
+  ac_cv_lib_m_sqrt=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_m_sqrt" >&5
+$as_echo "$ac_cv_lib_m_sqrt" >&6; }
+if test "x$ac_cv_lib_m_sqrt" = xyes; then :
+  cat >>confdefs.h <<_ACEOF
+#define HAVE_LIBM 1
+_ACEOF
+
+  LIBS="-lm $LIBS"
+
+fi
+
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for an ANSI C-conforming const" >&5
 $as_echo_n "checking for an ANSI C-conforming const... " >&6; }
@@ -15246,7 +15291,7 @@ else
   enable_werror=no
 fi
 
- if test "$enable_werror" == "yes"; then
+ if test "$enable_werror" = "yes"; then
   ENABLE_WERROR_TRUE=
   ENABLE_WERROR_FALSE='#'
 else
@@ -15255,7 +15300,7 @@ else
 fi
 
 
-if test "$enable_werror" == "yes"; then :
+if test "$enable_werror" = "yes"; then :
   WERROR_CFLAGS="-Werror"
 
 fi
@@ -15267,7 +15312,7 @@ else
   enable_coverage=no
 fi
 
- if test "$enable_coverage" == "yes"; then
+ if test "$enable_coverage" = "yes"; then
   ENABLE_COVERAGE_TRUE=
   ENABLE_COVERAGE_FALSE='#'
 else
@@ -15276,7 +15321,7 @@ else
 fi
 
 
-if test "$enable_coverage" == "yes"; then :
+if test "$enable_coverage" = "yes"; then :
   COVERAGE_CFLAGS="-fprofile-arcs -ftest-coverage"
 	COVERAGE_LIBS="-lgcov"
 
@@ -15380,7 +15425,7 @@ else
   enable_valgrind=no
 fi
 
- if test "$enable_valgrind" == "yes"; then
+ if test "$enable_valgrind" = "yes"; then
   ENABLE_VALGRIND_TRUE=
   ENABLE_VALGRIND_FALSE='#'
 else
@@ -15389,7 +15434,7 @@ else
 fi
 
 
-if test "$enable_valgrind" == "yes"; then :
+if test "$enable_valgrind" = "yes"; then :
   # Extract the first word of "valgrind", so it can be a program name with args.
 set dummy valgrind; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
